src/server/pfs/server/driver.go

```diff
@@ -1404,7 +1412,11 @@ func (d *driver) deleteCommit(ctx context.Context, commit *pfs.Commit) error {
 
 	for _, branchInfo := range branchInfos {
 		if branchInfo.Head != nil && branchInfo.Head.ID == commitInfo.Commit.ID {
-			if err := d.createBranch(ctx, branchInfo.Branch, commitInfo.ParentCommit, nil); err != nil {
+			var provenance []*pfs.Branch
+			for _, provBranch := range branchInfo.DirectProvenance {
+				provenance = append(provenance, provBranch)
+			}
+			if err := d.createBranch(ctx, branchInfo.Branch, commitInfo.ParentCommit, provenance); err != nil {
```

Comment:
So this is actually using `branchInfo.DirectProvenance` to assemble the `provenance` argument we pass to `createBranch`. We do this because we need to call `createBranch` to reset the head of the branch to `commitInfo.ParentCommit`, but we don't want this to clear the provenance. Previously, `createBranch` couldn't be used to remove provenance, only to add provenance, I fixed that as part of this PR but then had a bug where deleting a commit would leave a branch with no provenance and thus would prevent triggering.

All this being said, we may be able to remove this given that I think we agreed in the delete commit PR that you can't delete commits that have provenance, in which case provenance would always be `nil` here.